### PR TITLE
feat: Redirects "Soul Point Lost" messages to Overlay

### PR DESCRIPTION
**Pull request by san7890**
image
image

Another small QoL thing that extends the soulPoint configurable action.

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java [hunk 1]
```diff
@@ -587,6 +588,25 @@ protected String getNotification(Matcher matcher) {
         }
     }
 
+    private class ScrollTeleportationSoulPointRedirector extends SimpleRedirector {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java]: Can't we do this for x soul points? Like when we die, isn't there a 2 soul points lost message?

That would also mean renaming the redirector, adding a capture group to the regex, and adding a new config option

Comment (san7890) [reply common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java]: image

It's a completely different message (actually two messages) with different chatformatting codes. I can do that in another PR. You could also genericize the regex to account for either the dark_red/gray codes (also the has/have distinction)... but I think I'd rather that just be its own separate deal.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java]: The soul point part is almost the same? You can use * for the color code, and s? to fix the pluralization.

You can leave the you have died part in the chat

File: common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java [hunk 2]
```diff
@@ -641,6 +642,35 @@ protected String getNotification(Matcher matcher) {
         }
     }
 
+    private class SoulPointLossRedirector extends SimpleRedirector {
+        private static final Pattern SYSTEM_PATTERN =
+                Pattern.compile("^§[47](\\d+) soul points? (has|have) been lost...");
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java]:
```suggestion
                Pattern.compile("^§[47](\\d+) soul points? (has|have) been lost...$");
```

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**